### ISSUE TITLE
Add support for nanorc

### ISSUE THREAD

**Issue by perlun** [github/linguist#4335]
## Preliminary Steps

Please confirm you have...
- [X] reviewed [How Linguist Works](https://github.com/github/linguist#how-linguist-works),
- [X] reviewed the [Troubleshooting](https://github.com/github/linguist#troubleshooting) docs,
- [X] considered implementing an [override](https://github.com/github/linguist#overrides),
- [X] verified an issue has not already been logged for your issue ([linguist issues](https://github.com/issues?utf8=%E2%9C%93&q=is%3Aissue+repo%3Agithub/linguist)).

## Problem Description

[nano](https://www.nano-editor.org/) is a very commonly available console-based text editor for GNU/Linux and other systems. It's often installed by default, or easily installable using `apt-get` or similar mechanisms. Its configuration file is called `~/.nanorc`.

It would be great if linguist would provide support (syntax highlighting) for `nanorc` files. This issue exists to track adding this support.

Here is a reasonably complete syntax definition that could be used if anyone feels like looking into this: https://github.com/perlun/dotfiles/blob/master/nano/syntax/nanorc.nanorc (this file in itself uses the `nanorc` syntax, so it can also be used as an example)

### URL of the affected repository:
Example file: https://github.com/perlun/dotfiles/blob/master/nanorc

### Last modified on:
2018-11-22

### Expected language:
nanorc

### Detected language:
None

**Comment by Alhadis** [comment]
This would actually be a good idea. Dotfile repos are extremely common on GitHub, and many hackers tend to dig through them looking for tricks. Syntax highlighting for `.nanorc` files would certainly reach many users.

By the same token, other `.*rc` files would also benefit from syntax highlighting, such as [`.curlrc`](https://github.com/search?q=filename%3A.curlrc+NOT+nothack&type=Code). The only part that bothers me is that the addition would be purely aesthetic, and we don't even have highlighting/classification for `.gitignore` files (the single most common manifest file in Git repositories).

@pchaigno Any thoughts?

**Comment by pchaigno** [comment]
I'm all for it, but do we have a Sublime Text, Atom, or TextMate grammar we could use to highlight these files?

**Comment by perlun** [comment]
@pchaigno Maybe this could be of help?

- https://github.com/otgo/highlight-nanorc (2 years old, but nano probably hasn't changed that much since)

Looked for TextMate or Sublime but couldn't find any.

**Comment by Alhadis** [comment]
I'm tempted to hack one together, because we could leverage our [existing grammar](https://github.com/Alhadis/language-regexp) for highlighting regular expressions — all we'd need is a wrapper for it and some `.nanorc`-specific keyword patterns. 👍

The `otgo/highlight-nanorc` grammar [appears somewhat flaky](https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_format=cson&grammar_url=https%3A%2F%2Fraw.githubusercontent.com%2Fotgo%2Fhighlight-nanorc%2Fmaster%2Fgrammars%2Fnanorc.cson&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fgithub.com%2Fperlun%2Fdotfiles%2Fblob%2Fmaster%2Fnano%2Fsyntax%2Fnanorc.nanorc&code=). Notice the inconsistent highlighting.

**Comment by pchaigno** [comment]
> I'm tempted to hack one together

@Alhadis How much time would you need? Should we use `otgo/highlight-nanorc` in the meantime?

**Comment by Alhadis** [comment]
Already finished. Lol...

Haven't published it yet. I'll do so now.

**Comment by pchaigno** [comment]
@Alhadis If you want to submit the pull request to use your grammar, go ahead, I don't think @perlun is going to...

**Comment by Alhadis** [comment]
Will do. Should I add the grammars I wrote for `.gitignore`, `.gitattribute` and `.gitconfig` files too? They're, uh, damn near airtight in their syntax and tailored specifically to GitHub's colour scheme...

**Comment by pchaigno** [comment]
:+1: 

We should probably group the corresponding languages under a common parent language.

**Comment by Alhadis** [comment]
I guess `INI` will do. =)

**Comment by perlun** [comment]
> @Alhadis If you want to submit the pull request to use your grammar, go ahead, I don't think @perlun is going to...

Yeah, sorry, I didn't see this until now and I'm pretty certain @Alhadis is a more experienced contributor to the project anyway. =) So, I'm happy to delegate the work in this case. 😉 

Many thanks for looking into it! 👍
